src/Compilers/Core/Portable/Compilation/NullableContext.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the state of the nullable analysis at a specific point in a file.
+    /// </summary>
+    [Flags]
+    public enum NullableContext
+    {
+        /// <summary>
+        /// Nullable warnings and annotations are off at this location.
+        /// </summary>
+        Disabled = 0b00_00,
+
+        /// <summary>
+        /// The nullable warning state is inherited from the project default.
+        /// </summary>
+        /// <remarks>
+        /// The project default can change depending on the file type. Generated
+        /// files have nullable off by default, regardless of of the project-level
+        /// default setting.
+        /// </remarks>
+        WarningsContextInherited = 0b00_01,
+
+        /// <summary>
+        /// The nullable annotation tate is inherited from the project default.
+        /// </summary>
+        /// <remarks>
+        /// The project default can change depending on the file type. Generated
+        /// files have nullable off by default, regardless of of the project-level
+        /// default setting.
+        /// </remarks>
+        AnnotationsContextInherited = 0b00_10,
+
+        /// <summary>
+        /// The current state of both warnings and annotations are inherited from
+        /// the project default.
+        /// </summary>
+        /// <remarks>
+        /// This flag is set by default at the start of all files.
+        ///
+        /// The project default can change depending on the file type. Generated
+        /// files have nullable off by default, regardless of of the project-level
+        /// default setting.
+        /// </remarks>
+        ContextInherited = WarningsContextInherited | AnnotationsContextInherited,
+
+        /// <summary>
+        /// Nullable warnings are enabled and will be reported at this file location.
+        /// </summary>
+        WarningsEnabled = 0b01_00,
+
+        /// <summary>
+        /// Nullable annotations are enabled and will be shown when APIs defined at
+        /// this location are used in other contexts.
+        /// </summary>
+        AnnotationsEnabled = 0b10_00,
+
+        /// <summary>
+        /// The nullable feature is fully enabled.
+        /// </summary>
+        Enabled = WarningsEnabled | AnnotationsEnabled
+    }
+
+    public static class NullableContextExtensions
```

Comment:
Should there be an extension method to convert this to the NullableContext by throwing away the other bits (and having to shift them...)?

Reply:
I actually reordered the bits locally when I went to implement, so that the enable/disable bits lined up. As to whether there should be a public extension to convert them, I'd say let's hold off until we have a use case.